tools/closed-by-commits

```diff
@@ -0,0 +1,434 @@
+            )
+            all_seen_prs.add(pr.number)
+            pr_comments = pr_node.get("comments", {}).get("nodes", [])
+            for node in [pr_node, *pr_comments]:
+                pr.czo_urls.update(self._extract_czo_urls(node.get("body")))
+            pr.czo_urls.update(search_czo_for_number(self.czo_issue_prefix, pr.number))
+
+            # Get issues from PR metadata
+            issues_dict = {}
+            issue_nodes = pr_node.get("closingIssuesReferences", {}).get("nodes", [])
+            for issue_node in issue_nodes:
+                if issue_node is None:
+                    continue
+
+                issue = self._issue_from_node(issue_node)
+                issues_dict[issue.number] = issue
+
+            # Extract additional issue numbers from commit message
+            for issue_num in self._extract_issue_numbers(self.reponame, commit_message):
+                if issue_num not in issues_dict:
+                    maybe_issue = self._fetch_issue(issue_num)
+                    if maybe_issue is None:
+                        continue
+                    issues_dict[issue_num] = maybe_issue
+
+            for issue in issues_dict.values():
+                results[issue].add(pr)
+
+        if not pr_nodes:
+            for issue_num in self._extract_issue_numbers(self.reponame, commit_message):
+                maybe_issue = self._fetch_issue(issue_num)
+                if maybe_issue is None:
+                    continue
+                results[maybe_issue].update()
+
+        return results, all_seen_prs
+
+    def analyze_range(self, base: str, head: str) -> list[Issue]:
+        console = Console(stderr=True, log_path=False)
+
+        repository = self.github.get_repo(f"zulip/{self.reponame}")
+        comparison = repository.compare(base, head)
+        commit_shas = [commit.sha for commit in comparison.commits]
+        console.log(f"Found {len(commit_shas)} commits")
+
+        all_seen_prs = set()
+        issue_to_prs: dict[Issue, set[PullRequest]] = defaultdict(set)
+        with Progress(console=console) as progress:
+            task = progress.add_task("Processing commits...", total=len(commit_shas))
+            for sha in commit_shas:
+                progress.console.log(f"Processing {sha}")
+
+                this_issue_to_prs, this_seen_prs = self.get_issues_for_commit(sha)
+                for issue, prs in this_issue_to_prs.items():
+                    issue_to_prs[issue].update(prs)
+                all_seen_prs.update(this_seen_prs)
+                progress.advance(task)
+
+        for issue, prs in issue_to_prs.items():
+            for pr in prs:
+                issue.czo_urls.update(pr.czo_urls)
+
+            issue.closed_by_prs = sorted(prs, key=lambda p: p.number)
+
+        unique_prs = len({pr for prs in issue_to_prs.values() for pr in prs})
+        console.log(
+            f"Found {len(all_seen_prs)} PRs, with {unique_prs} unique PRs closing {len(issue_to_prs)} issues",
```

Comment:
nit: Perhaps:
```suggestion
            f"Found {len(all_seen_prs)} unique PRs, of which {unique_prs} PRs closing {len(issue_to_prs)} issues",
```

Otherwise it sounds like "unique" is a contrast between the two PR counts, but in fact both are counts of unique/distinct PRs.